tests/continuous/test_standard.py

```diff
@@ -912,6 +944,133 @@ def test_pbias_dask():
     xr.testing.assert_equal(result, EXP_PBIAS3)
 
 
+def test_within_no_data():
+
+    fcst = xr.DataArray(
+    np.array([np.nan]),
+    dims=("space"),
+    coords=[("space", ["x"])])
+
+    obs = xr.DataArray(
+    np.array([np.nan]),
+    dims=("space"),
+    coords=[("space", ["x"])])
+
+    result = scores.continuous.pwithin(
+        fcst,
+        obs,
+    )
+
+    assert result.ndim == 0
+    assert np.isnan(result.item())
+
+
+@pytest.mark.parametrize(
+    (
+        "fcst",
+        "obs",
+        "reduce_dims",
+        "preserve_dims",
+        "is_angular",
+        "threshold",
+        "rounded_digits",
+        "is_inclusive",
+        "expected",
+    ),
+    [
+        # Check reduce dim arg
+        (DA1_BIAS, DA2_BIAS, None, "space", False, 1.0, 7, True, EXP_PWITHIN1),
+        (DA1_BIAS, DA3_BIAS, None, "space", False, 1.0, 7, True, EXP_PWITHIN3),
+        # # Check preserve dim arg
+        (DA1_BIAS, DA2_BIAS, "time", None, False, 1.0, 7, True, EXP_PWITHIN1),
+        # Reduce all
+        (DA1_BIAS, DA2_BIAS, None, None, False, 1.0, 7, True, EXP_PWITHIN4),
+        # Test with Dataset
+        (DS_BIAS1, DS_BIAS2, None, "space", False, 1.0, 7, True, EXP_DS_PWITHIN1),
+        # Testing angular results
+        (DA1_ANGULAR, DA2_ANGULAR, None, None, True, 170.0, 7, True, EXP_PWITHIN5),
+        # Testing angular results with inclusive false
+        (DA1_ANGULAR, DA2_ANGULAR, None, None, True, 170.0, 7, False, EXP_PWITHIN6),
+    ],
+)
+def test_pwithin(fcst, obs, reduce_dims, preserve_dims, is_angular, threshold, rounded_digits, is_inclusive, expected):
+    """
+    Tests continuous.pwithin
+    """
+    result = scores.continuous.pwithin(
+        fcst,
+        obs,
+        reduce_dims=reduce_dims,
+        preserve_dims=preserve_dims,
+        is_angular=is_angular,
+        threshold=threshold,
+        rounded_digits=rounded_digits,
+        is_inclusive=is_inclusive,
+    )
+
+    xr.testing.assert_allclose(result, expected, rtol=1e-10, atol=1e-10)
+
+
+
+
+@pytest.mark.parametrize(
+    (
+        "fcst",
+        "obs",
+        "reduce_dims",
+        "preserve_dims",
+        "is_angular",
+        "threshold",
+        "rounded_digits",
+        "is_inclusive",
+        "expected",
+    ),
+    [
+        (DA1_BIAS, DA2_BIAS, None, None, False, 1.0, 2, True, 100 * 6 / 8),
+        (DA1_BIAS, DA2_BIAS, None, None, False, 1.0, 7, True, 100 * 4 / 8),
+    ],
+)
+def test_pwithin_tolerance(
+    fcst, obs, reduce_dims, preserve_dims, is_angular, threshold, rounded_digits, is_inclusive, expected
+):
+    """
+    Tests continuous.pwithin for threshold rounding to control for precision issues
+    """
+    fcst = fcst + 0.0001
+
+    result = scores.continuous.pwithin(
+        fcst,
+        obs,
+        reduce_dims=reduce_dims,
+        preserve_dims=preserve_dims,
+        is_angular=is_angular,
+        threshold=threshold,
+        rounded_digits=rounded_digits,
+        is_inclusive=is_inclusive,
+    )
```

Comment:
```suggestion
@pytest.mark.parametrize(
    (
        "rounded_digits",
        "expected",
    ),
    [
        (2, 100 * 6 / 8),
        (7, 100 * 4 / 8),
    ],
)
def test_pwithin_tolerance(rounded_digits, expected):
    """
    Tests continuous.pwithin for threshold rounding to control for precision issues
    """
    fcst = fcst + 0.0001

    result = scores.continuous.pwithin(
        DA1_BIAS,
        DA2_BIAS,
        threshold=1,
        rounded_digits=rounded_digits,
    )
```
This test can be simplified further since you were just repeating values in `pytest.mark.parametrize`- see suggested change